integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -135,6 +183,14 @@ def _convert_message_to_google_genai_format(message: ChatMessage) -> types.Conte
                     )
                 )
             )
+        elif isinstance(content_part, ReasoningContent):
+            # For assistant messages, check if they contain reasoning content that should be included
+            if message.is_from(ChatRole.ASSISTANT):
+                # Include thought content as a separate part to maintain context
+                thought_text = content_part.reasoning_text
+                if thought_text:
+                    # Create a thought part - this maintains the thought context
+                    parts.append(types.Part(text=thought_text, thought=True))
```

Comment:
1. Should thought summaries be included in subsequent calls? I cannot find evidence of this
2. (nit: this part can be simplified and some comments look redundant)